File: dillio-app/src/app/_models/product.ts
import { Image } from './../_model/image';

import { Category } from './category';


export interface Product{
    id?:number;
    price?:number;
    discount?:number;
    description?:string;
    title?:string;
    largeimageUrl?:string[];
    smallimageUrl?:string[];
    category?:Category;
}
export interface Spec {
  name: string;
  value: string;
  ProductId: number;

}
export interface IProduct {
  id: number;
  name: string;
  price: number;
  discount: number;
  categoryId: number;
  description: string;
  specs: Spec[];
  images: Image[];
}
export interface ProductResolved {
  product: IProduct;
  error?: any;
}


export interface IProductList {
  id: number;
  name: string;
  price: number;
  discount: number;
  description: string;
  categoryId: number;
  categoryName:string;
  images: Image[];
}

File: Dillio-Backend.DAL/Dillio-Backend.Entities/Core/Domain/Product.cs
using System;
using System.Collections.Generic;

namespace Dillio_Backend.BLL.Core.Domain
{
    [Serializable]
    public class Product
    {
        public Product()
        {
            Images = new HashSet<Image>();
            Reviews = new HashSet<Review>();
            Specs = new HashSet<Specs>();
        }
        public int Id { get; set; }
        public string Name { get; set; }
        public float Price { get; set; }
        public float Discount { get; set; }
        public string Description { get; set; }
        public virtual ICollection<Image> Images { get; set; }
        public virtual ICollection<Review> Reviews { get; set; }
        public virtual ICollection<Specs> Specs { get; set; }
        public virtual ICollection<ProductStore>ProductStores{ get; set; }
        public virtual Category Category { get; set; }
        public int CategoryId { get; set; }
    }
}
